core/trino-spi/src/main/java/io/trino/spi/SimdSupport.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi;
+
+/**
+ * Generic SIMD capability interface exposed by SPI.
+ * Implementations should answer true only when the operation is supported
+ * without emulation on the current CPU/JVM.
+ */
+public interface SimdSupport
```

Comment:
Instead of making this an interface with each CPU architecture defined separately, let's do CPU specific parsing and just emit a simple record class with fields for supported operations, e.g.:

```
public record SimdSupport(
   boolean nativeExpandAndCompressByte,
   boolean nativeExpandAndCompressShort,
   ...
)
```